adafruit_esp32spi/adafruit_esp32spi_socket.py

```diff
@@ -86,80 +86,15 @@ def send(self, data):  # pylint: disable=no-self-use
         _the_interface.socket_write(self._socknum, data, conn_mode=conntype)
         gc.collect()
 
-    def write(self, data):
-        """Sends data to the socket.
-        NOTE: This method is deprecated and will be removed.
-        """
-        self.send(data)
-
-    def readline(self, eol=b"\r\n"):
-        """Attempt to return as many bytes as we can up to but not including
-        end-of-line character (default is '\\r\\n')"""
-
-        # print("Socket readline")
-        stamp = time.monotonic()
-        while eol not in self._buffer:
-            # there's no line already in there, read some more
-            avail = self.available()
-            if avail:
-                self._buffer += _the_interface.socket_read(self._socknum, avail)
-            elif self._timeout > 0 and time.monotonic() - stamp > self._timeout:
-                self.close()  # Make sure to close socket so that we don't exhaust sockets.
-                raise RuntimeError("Didn't receive full response, failing out")
-        firstline, self._buffer = self._buffer.split(eol, 1)
-        gc.collect()
-        return firstline
-
-    def recv(self, bufsize=0):
+    def recv(self, bufsize: int) -> bytes:
         """Reads some bytes from the connected remote address. Will only return
         an empty string after the configured timeout.
 
         :param int bufsize: maximum number of bytes to receive
         """
-        # print("Socket read", bufsize)
-        if bufsize == 0:  # read as much as we can at the moment
```

Comment:
I made version that reverts this change for the special bufsize=0 behavior and was able to get an accurate image downloaded with that version. 

Perhaps something else is relying on this differing behavior somehow.

Reply:
That would make sense, though it's unfortunate, because the "read as much as possible" behavior is not what CPython does, assuming I am interpreting the doc correctly.

Reply:
Revisiting this, could we make socket the subset of CPython (which my understanding is the same as @dhalbert's that `0` would not be "read all"), and just have dependents have functionality for reading all?  Basically, move this up so that PortalBase would handle reading all available.

Reply:
@tekktrik I do think it should be possible to change the behavior in portalbase so that it can have the standard socket behavior here.